From fa38d6e60776888cca8ac296aca4f99ae4a4bbc1 Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Mon, 30 Nov 2020 21:26:10 +0200
Subject: [PATCH] [SW-20154] spi: change get_cs_pad_sel return type

u32 is defined in preboot.h and spi.h is included
as a header file in <HL_DEV>/common.h. In the next
patches, when base addresses of CFG/SPI/FIT etc are
moved into platform specific common.h, this would
create compile errors. Hence change return type
of get_cs_pad_sel to "unsigned int", so as to
prevent including preboot.h file

Change-Id: I232b58c4cc080b832df040476b9ed45a79680428
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/gaudi/common.c b/gaudi/common.c
index cfbee96..801c8c6 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -227,7 +227,7 @@
 	regmap_writel(1, mmPCIE_WRAP_VUART_TX_STATUS);
 }
 
-u32 get_cs_pad_sel(void)
+unsigned int get_cs_pad_sel(void)
 {
 	return mmPSOC_GLOBAL_CONF_PAD_SEL_53;
 }
diff --git a/goya/common.c b/goya/common.c
index 1cc3979..1a16d18 100644
--- a/goya/common.c
+++ b/goya/common.c
@@ -181,7 +181,7 @@
 	regmap_writel(1, mmPCIE_DB_RSV_RSV_1);
 }
 
-u32 get_cs_pad_sel(void)
+unsigned int get_cs_pad_sel(void)
 {
 	return mmPSOC_GLOBAL_CONF_PAD_SEL_53;
 }
diff --git a/goya2/common.c b/goya2/common.c
index 84fbe4b..2e775fc 100644
--- a/goya2/common.c
+++ b/goya2/common.c
@@ -320,7 +320,7 @@
 	regmap_writel(1, mmPCIE_WRAP_VUART_TX_STATUS);
 }
 
-u32 get_cs_pad_sel(void)
+unsigned int get_cs_pad_sel(void)
 {
 	return mmPSOC_GLOBAL_CONF_PAD_SEL_19;
 }
diff --git a/goya2_arc/common.c b/goya2_arc/common.c
index dcdb91c..c2f7020 100644
--- a/goya2_arc/common.c
+++ b/goya2_arc/common.c
@@ -203,7 +203,7 @@
 	regmap_writel(1, mmPCIE_WRAP_VUART_TX_STATUS);
 }
 
-u32 get_cs_pad_sel(void)
+unsigned int get_cs_pad_sel(void)
 {
 	return mmPSOC_GLOBAL_CONF_PAD_SEL_19;
 }
diff --git a/goya2_arc_gnu/common.c b/goya2_arc_gnu/common.c
index 0b7f89c..9776f96 100644
--- a/goya2_arc_gnu/common.c
+++ b/goya2_arc_gnu/common.c
@@ -118,7 +118,7 @@
 	regmap_writel(1, mmPCIE_WRAP_VUART_TX_STATUS);
 }
 
-u32 get_cs_pad_sel(void)
+unsigned int get_cs_pad_sel(void)
 {
 	return mmPSOC_GLOBAL_CONF_PAD_SEL_19;
 }
diff --git a/include/spi.h b/include/spi.h
index 28ea6a6..49ded97 100644
--- a/include/spi.h
+++ b/include/spi.h
@@ -26,6 +26,6 @@
 
 void spi_copy(void *dst, unsigned int offset, unsigned int size, char *name);
 void spi_copy_otp(void *dst, unsigned int offset, unsigned int size);
-u32 get_cs_pad_sel(void);
+unsigned int get_cs_pad_sel(void);
 
 #endif /* _HL_SPI_H_ */
